Add contributing guidelines and update README (#278)
mautic/developer-documentation-new: README.md

> # Mautic developer documentation (new)

Reviewer: ```suggestion
# Mautic developer documentation
```

I think we can remove the 'new' part now.

> This repository hosts the new developer documentation for Mautic on the [Read the Docs platform][ReadTheDocs]. Whenever a PR is merged, changes are deployed immediately to https://devdocs.mautic.org/.

Reviewer: ```suggestion
This repository hosts the new developer documentation for Mautic on the [Read the Docs platform][ReadTheDocs]. Builds take a few minutes to run before changes show on  https://devdocs.mautic.org following the merging of a PR. Each PR has a GitHub Action which spins up a temporary environment for reviewing the changes.
```

> If you're looking for the legacy developer documentation, please go to https://developer.mautic.org/ or the [GitHub repository][Legacy dev docs].

Reviewer: ```suggestion
If you're looking for the legacy developer documentation, please go to https://developer.mautic.org/ or the [GitHub repository][Legacy dev docs]. This still exists because we have some changes which haven't yet been ported over to the new documentation here - if you'd like to help, please check the [contribution guidelines](.github/CONTRIBUTING.md).
```

[ReadTheDocs]: <https://readthedocs.org>
[Legacy dev docs]: <https://github.com/mautic/developer-documentation>

## Contributing 🤝

> All kinds of contributions are encouraged. For complete information on how to contribute to the Mautic developer documentation (new) repository, please read the [Contributing Guidelines](.github/CONTRIBUTING.md).

Reviewer: ```suggestion
Mautic actively encourages all kinds of contributions. For complete information on how to contribute to the Mautic developer documentation repository, please read the [Contributing Guidelines](.github/CONTRIBUTING.md).
```

> All contributors are required to abide by our [Code of Conduct](https://mautic.org/code-of-conduct/).

Reviewer: ```suggestion
All contributors must abide by the Mautic [Code of Conduct](https://mautic.org/code-of-conduct/).
```
